File: trunk/src/main/resources/static/pc/trade/model/Trade.js
//定义ViewModel对象
var TradeEditViewModel = function () {  
	var self=this;
	self.tradeId = ko.observable(''); 
	self.serialNumber = ko.observable(''); 
	self.customerId = ko.observable(''); 
	self.customerCt = ko.observable(''); 
	self.authorizationId = ko.observable(''); 
	self.objectType = ko.observable(''); 
	self.objectId = ko.observable(''); 
	self.objectCt = ko.observable(''); 
	self.status = ko.observable(''); 
	self.note = ko.observable(''); 
	self.markedPrice = ko.observable(''); 
	self.discountPrice = ko.observable(''); 
	self.dealPrice = ko.observable(''); 
	self.couponId = ko.observable(''); 

    var opFalg=getQueryString('action');
    
    if(opFalg!="Add"){
    	var opid=getQueryString('id');
    	myAjax("/trade/"+opid, "GET", null, function (data){
			self.tradeId(data.tradeId);
			self.serialNumber(data.serialNumber);
			self.customerId(data.customerId);
			self.customerCt(data.customerCt);
			self.authorizationId(data.authorizationId);
			self.objectType(data.objectType);
			self.objectId(data.objectId);
			self.objectCt(data.objectCt);
			self.status(data.status);
			self.note(data.note);
			self.markedPrice(data.markedPrice);
			self.discountPrice(data.discountPrice);
			self.dealPrice(data.dealPrice);
			self.couponId(data.couponId);
		}, true);
	}

	//【提交】按钮押下处理
    self.Commit = function () {    
    	var submitPar ={};
		submitPar.serialNumber=self.serialNumber();
		submitPar.customerId=self.customerId();
		submitPar.customerCt=self.customerCt();
		submitPar.authorizationId=self.authorizationId();
		submitPar.objectType=self.objectType();
		submitPar.objectId=self.objectId();
		submitPar.objectCt=self.objectCt();
		submitPar.status=self.status();
		submitPar.note=self.note();
		submitPar.markedPrice=self.markedPrice();
		submitPar.discountPrice=self.discountPrice();
		submitPar.dealPrice=self.dealPrice();
		submitPar.couponId=self.couponId();
    	
    	if(opFalg=="Add"){
    		myAjaxJson("/trade", "POST", submitPar, function (data){
				ChangeUrl("./trade/TradeList.html");
			}, true);
		}else{
    		var opid=getQueryString('id');
    		myAjaxJson("/trade/"+opid, "PUT", submitPar, function (data){
				ChangeUrl("./trade/TradeList.html");
			}, true);
    	}
    };
};


$().ready(function(){
	$("#txtName").focus();
    ko.applyBindings(new TradeEditViewModel());
});

File: trunk/src/main/java/com/jing/trade/model/entity/Trade.java
package com.jing.trade.model.entity;

import com.jing.utils.BaseEntity;

/**
 * 消费清单 实体类
 * 
 * @author codeing gen
 */
public class Trade extends BaseEntity {
	private static final long serialVersionUID = 1L;

	private Integer tradeId; // 交易标识
	private String serialNumber; // 流水号
	private Integer customerId; // 客户标识
	private Integer customerCt; // 客人数
	private String authorizationId; // 授权码标识
	private Integer objectType; // 商品服务类型
	private String objectId; // 商品服务标识
	private Integer objectCt; // 交易数量
	private Integer status; // 清单状态 0正常
	private String note; // 交易说明
	private java.math.BigDecimal markedPrice; // 原价
	private java.math.BigDecimal discountPrice; // 优惠金额
	private java.math.BigDecimal dealPrice; // 成交价
	private Integer couponId; // 活动标识

	public Integer getTradeId() {
		return this.tradeId;
	}

	public String getSerialNumber() {
		return this.serialNumber;
	}

	public Integer getCustomerId() {
		return this.customerId;
	}

	public Integer getCustomerCt() {
		return this.customerCt;
	}

	public String getAuthorizationId() {
		return this.authorizationId;
	}

	public Integer getObjectType() {
		return this.objectType;
	}

	public String getObjectId() {
		return this.objectId;
	}

	public Integer getObjectCt() {
		return this.objectCt;
	}

	public Integer getStatus() {
		return this.status;
	}

	public String getNote() {
		return this.note;
	}

	public java.math.BigDecimal getMarkedPrice() {
		return this.markedPrice;
	}

	public java.math.BigDecimal getDiscountPrice() {
		return this.discountPrice;
	}

	public java.math.BigDecimal getDealPrice() {
		return this.dealPrice;
	}

	public Integer getCouponId() {
		return this.couponId;
	}

	public void setTradeId(Integer tradeId) {
		this.tradeId = tradeId;
	}

	public void setSerialNumber(String serialNumber) {
		this.serialNumber = serialNumber;
	}

	public void setCustomerId(Integer customerId) {
		this.customerId = customerId;
	}

	public void setCustomerCt(Integer customerCt) {
		this.customerCt = customerCt;
	}

	public void setAuthorizationId(String authorizationId) {
		this.authorizationId = authorizationId;
	}

	public void setObjectType(Integer objectType) {
		this.objectType = objectType;
	}

	public void setObjectId(String objectId) {
		this.objectId = objectId;
	}

	public void setObjectCt(Integer objectCt) {
		this.objectCt = objectCt;
	}

	public void setStatus(Integer status) {
		this.status = status;
	}

	public void setNote(String note) {
		this.note = note;
	}

	public void setMarkedPrice(java.math.BigDecimal markedPrice) {
		this.markedPrice = markedPrice;
	}

	public void setDiscountPrice(java.math.BigDecimal discountPrice) {
		this.discountPrice = discountPrice;
	}

	public void setDealPrice(java.math.BigDecimal dealPrice) {
		this.dealPrice = dealPrice;
	}

	public void setCouponId(Integer couponId) {
		this.couponId = couponId;
	}

}